products/swift-bindings/CMakeLists.txt

```diff
@@ -1,2 +1,83 @@
-add_custom_target(swift-bindings
-                  DEPENDS libllbuild)
+# FIXME: This function should probably be in cmake/modules/Utility.cmake
+# FIXME: Incremental builds.  
+# Compile swift sources to swiftmodules.
+# Usage:
+# target          # Target name
+# name            # Swift Module name
+# deps            # Target dependencies
+# sources         # List of sources
+# additional_args # List of additional args to pass
+function(add_swift_module target name deps sources additional_args)
+
+  list(APPEND ARGS -module-name)
+  list(APPEND ARGS ${name})
+  list(APPEND ARGS -incremental -emit-dependencies)
+  list(APPEND ARGS -emit-module)
+  list(APPEND ARGS -emit-module-path)
+  list(APPEND ARGS ${name}.swiftmodule)
+  
+  set(BUILD_DIR ${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY})
+  set(FILEMAP ${BUILD_DIR}/output-file-map.json)
+  
+  file(REMOVE ${FILEMAP})
+  file(APPEND ${FILEMAP} "{\n\"\": {\n")
+  file(APPEND ${FILEMAP} "\"swift-dependencies\": \"${BUILD_DIR}/master.swiftdeps\"\n")
+  file(APPEND ${FILEMAP} "},\n")
+  
+  foreach(source ${sources})
+    file(APPEND ${FILEMAP} "\"${CMAKE_CURRENT_SOURCE_DIR}/${source}\": {\n")
+    file(APPEND ${FILEMAP} "\"dependencies\": \"${BUILD_DIR}/${source}.d\",\n")
+    file(APPEND ${FILEMAP} "\"object\": \"${BUILD_DIR}/${source}.o\",\n")
+    file(APPEND ${FILEMAP} "\"swiftmodule\": \"${BUILD_DIR}/${source}~partial.swiftmodule\",\n")
+    file(APPEND ${FILEMAP} "\"swift-dependencies\": \"${BUILD_DIR}/${source}.swiftdeps\"\n}\n")
+    # FIXME: add comma for all but last source.
+  endforeach()
+  file(APPEND ${FILEMAP} "}")
+    
+  list(APPEND ARGS -output-file-map ${BUILD_DIR}/output-file-map.json)
+  list(APPEND ARGS -parse-as-library)
+  list(APPEND ARGS -c)
+
+  foreach(source ${sources})
+      list(APPEND ARGS ${CMAKE_CURRENT_SOURCE_DIR}/${source})
+  endforeach()
+
+  if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+      list(APPEND ARGS -Onone -g)
+  else()
+      list(APPEND ARGS -O2)
+  endif()
+
+  list(APPEND ARGS -D SWIFT_PACKAGE)
+
+  foreach(arg ${additional_args})
+      list(APPEND ARGS ${arg})
+  endforeach()
+  
+  set(SWIFT_EXEC swiftc)
+
+  # FIXME: Get proper sdk paths.
+  if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    set(SWIFT_EXEC xcrun -sdk macosx swiftc)
+  endif()
+  
+  add_custom_command(
+      OUTPUT    ${name}
+      COMMAND   ${SWIFT_EXEC}
+      ARGS    ${ARGS}
+  )
+  add_custom_target(${target} ALL DEPENDS ${deps} ${name})
+endfunction()
+
+# Set sources.
+set(SOURCES llbuild.swift)
+
+# Link C API.
+list(APPEND addtional_args -lllbuild)
+list(APPEND addtional_args -Xcc -fmodule-map-file=${CMAKE_CURRENT_SOURCE_DIR}/../libllbuild/public-api/llbuild/module.modulemap)
```

Comment:
This doesn't need to actually use `-fmodule-map-file`, it can just pass `-I path/to/public-api/llbuild` and Clang will automatically find the module map.

BTW, it should, I believe, use `-import-underlying-module` to cause the `llbuild` module to be imported (because we are going to want the Swift bindings to also be called "llbuild", I think.